Motion tune

jr-brown opened the pull request.

jr-brown committed:
Collision avoidance using new target_pos

WeixuanZ committed:
Change collision avoidance target pos sign

jr-brown committed:
Overhaul debugging

WeixuanZ committed:
Revert "Collision avoidance using new target_pos"

This reverts commit 1a2d72318bf604314d660cec08a9b3cb82b9a580.

WeixuanZ committed:
Revert "Merge branch 'adv-collision-avoidance' of https://github.com/IDP-L211/controllers into adv-collision-avoidance"

This reverts commit e88581628b9882db4967dadfccae66223669882a, reversing
changes made to 89ca21be53f1de3a231fde59313648e3144a2fde.

WeixuanZ committed:
Add back debug flag change

jr-brown committed:
Tune PID and motion numbers

jr-brown committed:
Changing collect to make it more robust

jr-brown committed:
Add move to block with offset

jr-brown committed:
More robust PID and add easier graph plotting

jr-brown committed:
Add logic for collecting far targets

jr-brown committed:
Passive collision avoidance changes angle not r_speed

jr-brown committed:
Add non-linear angle controller

jr-brown committed:
Add variable accuracy for drive to position and more debug options

jr-brown committed:
FIx collision avoidance and debug

jr-brown committed:
Add stuck detection into drive_to_position

jr-brown committed:
New passive collision avoidance

jr-brown committed:
Adjust distance for new sensors

jr-brown committed:
Square scan fallback strategy

jr-brown committed:
Better collision avoidance with other bot

Note no passive collision avoidance with blocks

WeixuanZ committed:
[WIP] Robust collision avoidance (#28)

* Initial implementation of passive collision avoidance

* Better rotation control; remove get out of stuck

* Do not pass data points likely to be robot into DBSCAN

* Deal with flipped blocks

* Different distance to block to collect

* Overhaul debugging

Co-authored-by: jasonb <jasonrbrown@virginmedia.com>
Co-authored-by: Tim Clifford <tclifford@protonmail.com>

jr-brown committed:
Fixup

jr-brown committed:
Add spin detection and add back old scan fallback algorithm

jr-brown committed:
Put move tolerance back

Really sorry for the commit spam lol rip your inboxes

jr-brown committed:
Collision avoidance of scanned blocks

jr-brown committed:
Collision avoidance of groups of nearby obstructions

eleanor-clifford reviewed (commented):
Smh octo failed and I have to do the final bit on the website :angry:

jr-brown committed:
Make passive collision avoidance its own function

jr-brown committed:
Color detection over a range of distances

WeixuanZ committed:
Increase path rectangle pass the target

jr-brown committed:
Fix continuous color detection

jr-brown committed:
Dont mutate position when detecting color

jr-brown committed:
Dont mutate position in approach

Also use a different variable for spin stuck detection than the one used
for rotations

jr-brown committed:
Fix a horrible bug in rotate to face block

jr-brown committed:
Reduce collection rotation when other blocks are nearby

jr-brown committed:
Improve protocol when other blocks are near our target

jr-brown committed:
Small tweaks to values

jr-brown committed:
Add passive collision avoidance for drive to color detect start

jr-brown committed:
Improve bot communication

jr-brown committed:
Bots communicate disappeared targets

jr-brown committed:
Bots don't have same target and other changes

Made color detection symetric

Increased threshold for considering block detections equal

jr-brown committed:
Update motion characteristics and rotation after approach

jr-brown committed:
Increase rotate angle when blocks nearby

jr-brown committed:
Avoid target passively on approach

jr-brown committed:
Emergency collision avoidance protocol
